chalicelib/request_handler.py

```diff
@@ -0,0 +1,94 @@
+import hashlib
+import json
+import os
+import tempfile
+import boto3
+
+from enum import Enum
+from botocore.exceptions import ClientError
+from chalicelib.constants import MERGED_REQUEST_STATUS_BUCKET_NAME, JSON_EXTENSION, REQUEST_TEMPLATE_PATH, \
+    MERGED_MTX_BUCKET_NAME
+
+
+class RequestStatus(Enum):
+    UNINITIALIZED = "UNINITIALIZED"
+    RUNNING = "RUNNING"
+    DONE = "DONE"
+
+
+class RequestHandler:
+    @staticmethod
+    def generate_request_id(bundle_uuids):
+        """
+        Generate a request id based on a list of bundle uuids.
+        :param bundle_uuids: A list of bundle uuids.
+        :return: Request id.
+        """
+        bundle_uuids.sort()
+        m = hashlib.sha256()
+
+        for bundle_uuid in bundle_uuids:
+            m.update(bundle_uuid.encode())
+
+        return m.hexdigest()
+
+    @staticmethod
+    def check_request_status(request_id):
+        """
+        Check the status of a matrix concatenation request.
+        :param request_id: Matrices concatenation request id.
+        :return: The status of the request.
+        """
+        s3 = boto3.resource("s3")
+        key = request_id + JSON_EXTENSION
+
+        try:
+            response = s3.Object(bucket_name=MERGED_REQUEST_STATUS_BUCKET_NAME, key=key).get()
+            body = json.loads(response['Body'].read())
+            return RequestStatus(body["status"])
+
+        except ClientError as e:
+            if e.response['Error']['Code'] == "NoSuchKey":
+                return RequestStatus("UNINITIALIZED")
+            else:
+                raise e
+
+    @staticmethod
+    def update_request(bundle_uuids, request_id, status):
+        """
+        Create a request status json file in s3 bucket.
+        :param bundle_uuids: A list of bundle uuids.
+        :param request_id: Matrices concatenation request id.
+        :param status: Request status to update.
+        """
+        with open(REQUEST_TEMPLATE_PATH) as f:
+            # Create a request based on a template json file
+            request = json.load(f)
+            request["bundle_uuids"] = bundle_uuids
+            request["status"] = status
+            request["request_id"] = request_id
+
+            if status == RequestStatus.DONE.name:
```

Comment:
It might be simpler simply to pass in and compare the enum itself, rather than the enum's name.

Reply:
If I directly pass in an enum object into the json, then I will have to do work to serialize/deserialize the enum. Is it worth to do that instead of directly passing the enum's name, which is simpler?

Reply:
(Discussed this in person)